cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -92,18 +84,58 @@ type certManagerClient struct {
 	configClient        config.Client
 	proxy               Proxy
 	pollImmediateWaiter PollImmediateWaiter
+	embeddedCertManagerManifestVersion string
+	embeddedCertManagerManifestHash string
 }
 
 // Ensure certManagerClient implements the CertManagerClient interface.
 var _ CertManagerClient = &certManagerClient{}
 
+func getManifestHash() string {
+	yaml, err := manifests.Asset(embeddedCertManagerManifestPath)
+        if err != nil {
+                return ""
+        }
+       	return fmt.Sprintf("%x", sha256.Sum256(yaml))
+}
+
+// Images return the list of images required for installing the cert-manager.
+func setManifestLabel(cm *certManagerClient) error {
+        // Gets the cert-manager objects from the embedded assets.
+        objs, err := cm.getManifestObjs()
+        if err != nil {
+                return err
+        }
+
+	for i := range objs {
+                o := objs[i]
+                if o.GetKind() == "CustomResourceDefinition" {
+                        labels := o.GetLabels()
+                        version, ok := labels[certmanagerVersionLable]
+                        if ok {
+				s := strings.Split(version, "-")			
+                                cm.embeddedCertManagerManifestVersion = s[2]
+				break
+                        }
```

Comment:
I think we should return an error in case the label does not exist and/or in the case the label is not in the expected format (e.g. s have a different length than expected)

Reply:
let me think about it ...thanks for the reminder